### PR TITLE
GFORMS-1786: DependencyGraph - handle references to atl field from outside atl

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/eval/ExpressionResult.scala [hunk 1]
```diff
@@ -126,8 +126,8 @@ sealed trait ExpressionResult extends Product with Serializable {
     case t: Invalid      => false
     case t: Hidden.type  => false
     case t: Empty.type   => false
-    case t: NumberResult => false
-    case t: StringResult => false
+    case t: NumberResult => identical(er)
+    case t: StringResult => identical(er)
```
[the remark this answers is not in the document]
Comment (lxol) [reply app/uk/gov/hmrc/gform/eval/ExpressionResult.scala:130]: not directly related to this ticket but related to fixing Ratheesh's failed form, where we changed `==` to `contains`.

[the remark this answers is not in the document]
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/eval/ExpressionResult.scala:130]: In that case this is not a proper fix a should be reverted here and fixed elsewhere

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/sharedmodel/graph/DependencyGraph.scala [hunk 2]
```diff
@@ -147,28 +144,32 @@ object DependencyGraph {
...
-              }
-              .map { v =>
-                GraphNode.Simple(ModelComponentId.pure(IndexedComponentId.pure(k)).toFormComponentId) ~> GraphNode.Expr(
-                  FormCtx(ModelComponentId.pure(v).toFormComponentId)
-                )
-              }
+        .collect { case (k, fcs) =>
+          fcs.flatMap { fc =>
+            (fc match {
+              case f @ IsRevealingChoice(revealingChoice) if atlFieldsBaseIds(k) =>
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/graph/DependencyGraph.scala:159]: This special `IsRevealingChoice` case is awkward.
I have provided slightly different implementation [here](https://github.com/hmrc/gform-frontend/commit/ce528a587faf0abfca6d56849162ae9266b1f448)

Comment (lxol) [reply app/uk/gov/hmrc/gform/sharedmodel/graph/DependencyGraph.scala]: I cherry-picked your PR. :+1:

**Pull request merged**